src/github/pages/README.md

```diff
@@ -0,0 +1,45 @@
+# `garden github-pages`
+
+Publish the given directory to a `gh-pages` branch.
+
+## Usage
+
+```ts
+import { githubPages } from '@zendeskgarden/tool-shed';
+
+const args: {
+  dir: string;
+  path?: string;
+  message?: string;
+  token?: string;
+} = {
+  dir: __dirname
+  /* optional overrides */
+};
+
+(async () => {
+  const url = await githubPages(args);
+
+  console.log(url);
+})();
+```
+
+### Arguments
+
+- `dir` directory of web content to publish.
+- `path` optional path to a git directory; defaults to `dir`.
+- `message` optional commit message.
+- `token` optional GitHub personal access token; defaults to the value
+  provided by [`githubToken`](../token#readme).
+
+## Command
+
+```sh
+garden github-pages [--path <path>] [--message <message>] [--token <token>] <dir>
```

Comment:
This can be used for any repo that publishes to a `gh-pages` branch. The logic here consolidates repeated Garden configurations under command defaults so that we stop repeating ourselves.